js/views/modals/purchase/Shipping.js

```diff
@@ -0,0 +1,43 @@
+import '../../../lib/select2';
+import app from '../../../app';
+import loadTemplate from '../../../utils/loadTemplate';
+import BaseModal from '../BaseModal';
+import Listing from '../../../models/listing/Listing';
+
+export default class extends BaseModal {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    if (!this.model || !(this.model instanceof Listing)) {
+      throw new Error('Please provide a listing model');
+    }
+
+    this.listenTo(app.settings.get('shippingAddresses'), 'update',
+      (cl, updateOpts) => {
+        if (updateOpts.changes.added.length ||
+          updateOpts.changes.removed.length) {
```

Comment:
As an FYI...this condition will account for a shippingAddress being added or removed. If you also want to account for an existing shippingAddress changing... I believe that information would be in `updateOpts.changes.merged`. It may be easier to just re-render the view on update depending on how the view keeps track of it's own state and how grace-fully it re-renders (it also may be easier to do the updates with jQuery - all depends on how you architect the view).